### ISSUE TITLE
SynchronizationLockException in DefaultHttpClientFactory.CleanupTimer_Tick

### ISSUE THREAD

**Issue by zone117x** [aspnet/HttpClientFactory#140]
Came across a threading bug when running a long test suite on a slow CI server. Using xuint, netcoreapp2.1, and ubuntu 16.04 x64.

```
[18:36:59]	[test] The active test run was aborted. Reason: Unhandled Exception: Unhandled Exception: 
[18:36:59]	[test] Unhandled Exception: Unhandled Exception: System.Threading.SynchronizationLockException: Object synchronization method was called from an unsynchronized block of code.
[18:36:59]	[test]    at System.Threading.Monitor.Exit(Object obj)
[18:36:59]	[test]    at Microsoft.Extensions.Http.DefaultHttpClientFactory.CleanupTimer_Tick(Object state)
[18:36:59]	[test]    at System.Threading.ExecutionContext.RunInternal(ExecutionContext executionContext, ContextCallback callback, Object state)
[18:36:59]	[test] --- End of stack trace from previous location where exception was thrown ---
[18:36:59]	[test]    at System.Threading.TimerQueueTimer.CallCallback()
[18:36:59]	[test]    at System.Threading.TimerQueueTimer.Fire()
[18:36:59]	[test]    at System.Threading.TimerQueue.FireNextTimers()System.Threading.SynchronizationLockException: Object synchronization method was called from an unsynchronized block of code.
[18:36:59]	[test]    at System.Threading.Monitor.Exit(Object obj)
[18:36:59]	[test]    at Microsoft.Extensions.Http.DefaultHttpClientFactory.CleanupTimer_Tick(Object state)
[18:36:59]	[test]    at System.Threading.ExecutionContext.RunInternal(ExecutionContext executionContext, ContextCallback callback, Object state)
[18:36:59]	[test] --- End of stack trace from previous location where exception was thrown ---
[18:36:59]	[test]    at System.Threading.TimerQueueTimer.CallCallback()
[18:36:59]	[test]    at System.Threading.TimerQueueTimer.Fire()
[18:36:59]	[test]    at System.Threading.TimerQueue.FireNextTimers()
[18:36:59]	[test] System.Threading.SynchronizationLockException: Object synchronization method was called from an unsynchronized block of code.
[18:36:59]	[test]    at System.Threading.Monitor.Exit(Object obj)
[18:36:59]	[test]    at Microsoft.Extensions.Http.DefaultHttpClientFactory.CleanupTimer_Tick(Object state)
[18:36:59]	[test]    at System.Threading.ExecutionContext.RunInternal(ExecutionContext executionContext, ContextCallback callback, Object state)
[18:36:59]	[test] --- End of stack trace from previous location where exception was thrown ---
[18:36:59]	[test]    at System.Threading.TimerQueueTimer.CallCallback()
[18:36:59]	[test]    at System.Threading.TimerQueueTimer.Fire()
[18:36:59]	[test]    at System.Threading.TimerQueue.FireNextTimers()
[18:36:59]	[test] System.Threading.SynchronizationLockException: Object synchronization method was called from an unsynchronized block of code.
[18:36:59]	[test]    at System.Threading.Monitor.Exit(Object obj)
[18:36:59]	[test]    at Microsoft.Extensions.Http.DefaultHttpClientFactory.CleanupTimer_Tick(Object state)
[18:36:59]	[test]    at System.Threading.ExecutionContext.RunInternal(ExecutionContext executionContext, ContextCallback callback, Object state)
[18:36:59]	[test] --- End of stack trace from previous location where exception was thrown ---
[18:36:59]	[test]    at System.Threading.TimerQueueTimer.CallCallback()
[18:36:59]	[test]    at System.Threading.TimerQueueTimer.Fire()
[18:36:59]	[test]    at System.Threading.TimerQueue.FireNextTimers()
[18:36:59]	[test] 
[18:37:00]	[test] Test Run Aborted.
```

**Comment by rynowak** [comment]
Thanks, this looks like something we need to dig into.

**Comment by rynowak** [comment]
Fixed by #153  - the PR has the wrong issue reference
